### PR TITLE
Use namespaced liblcf (depends on liblcf#361)

### PR DESCRIPTION
**Pull request by fmatthew5876**
Depends on https://github.com/EasyRPG/liblcf/pull/361

### CODE REVIEW

**Comment by fmatthew5876** [comment]
Rebased onto fmt.

Checked `ag Output:: | grep %`. Compiles on lcf branch

**Comment by Ghabry** [comment]
is there any working clang tool available again that can cleanup/resort the include list? Because liblcf is now \<\> it doesn't fit optical in the include list anymore (but this can be resolved later)

**Comment by fmatthew5876** [comment]
`clang-format` can do this. It could also be used if we want to enforce a style guide later on or as part of a commit hook to check whitespace etc.. etc..

File: src/sdl2_ui.cpp [hunk 1]
```diff
@@ -301,7 +301,7 @@ bool Sdl2Ui::RefreshDisplayMode() {
 			return false;
 		}
 
-		auto renderer_sg = makeScopeGuard([&]() {
+		auto renderer_sg = lcf::makeScopeGuard([&]() {
```
Comment (Ghabry) [inline src/sdl2_ui.cpp:304]: The lcf-namespace nicely indicates now that scope guard doesn't really fit there - A liblcf dependency in SDL2_Ui :thinking:

Comment (fmatthew5876) [reply src/sdl2_ui.cpp:304]: Yep, it's ugly. But the alternative is another repo and then we have 3 to keep in sync 😭 .

**Comment by Ghabry** [comment]
Do you have any search-replace list or a script for us to automate this boring task when fixing tools and editor?

**Comment by fmatthew5876** [comment]
`grep`, `ag`, and `sed` is what I used 😄

**Comment by carstene1ns** [comment]
jenkins: test this please.

**Comment by carstene1ns** [comment]
hello mr. jenkins, test this please!

File: src/player.cpp [hunk 2]
```diff
@@ -848,8 +848,8 @@ void Player::ResetGameObjects() {
 }
 
 void Player::LoadDatabase() {
-	// Load Database
-	Data::Clear();
+	// Load lcf::Database
```
Comment (carstene1ns) [inline src/player.cpp:851]: Accidental comment change, but superfluous comment anyway.

File: src/sdl2_ui.cpp (hunk 1 above)
Comment (carstene1ns) [reply src/sdl2_ui.cpp:304]: Also thought about having this in a git subtree. "inside, but outside"

File: src/game_interpreter.cpp [hunk 3]
```diff
@@ -786,7 +786,7 @@ bool Game_Interpreter::CommandShowMessage(RPG::EventCommand const& com) { // cod
 	++index;
 
 	// Check for continued lines via ShowMessage_2
-	while (index < static_cast<int>(list.size()) && list[index].code == Cmd::ShowMessage_2) {
+	while (index < static_cast<int>(list.size()) && static_cast<Cmd>(list[index].code) == Cmd::ShowMessage_2) {
```
Comment (carstene1ns) [inline src/game_interpreter.cpp:789]: Maybe we can resolve all these added `static_cast`s in another way after merging?

Comment (fmatthew5876) [reply src/game_interpreter.cpp:789]: https://github.com/EasyRPG/liblcf/issues/370

**Comment by carstene1ns** [comment]
alllllllrighty then!

**Pull request merged**